sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -67,6 +73,373 @@ describe(
       );
     });
 
+    it("Should create a notification for the ministry and student for a blocked disbursement when there are no previously existing notifications for the disbursement.", async () => {
+      // Arrange
+      // Student with invalid SIN to block the disbursement.
+      const student = await saveFakeStudent(db.dataSource, undefined, {
+        sinValidationInitialValue: {
+          isValidSIN: false,
+        },
+      });
+      // Valid MSFAA Number.
+      const msfaaNumber = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.Signed,
+            msfaaInitialValues: {
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
+      // Student application.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student, msfaaNumber },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Completed,
+          currentAssessmentInitialValues: {
+            assessmentData: { weeks: 5 } as Assessment,
+            assessmentDate: new Date(),
+          },
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+          },
+        },
+      );
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processECert(mockedJob.job);
+
+      // Assert
+      expect(result).toStrictEqual([
+        "Process finalized with success.",
+        "Generated file: none",
+        "Uploaded records: 0",
+      ]);
+      expect(
+        mockedJob.containLogMessages([
+          `Creating notifications for disbursement id: ${application.currentAssessment.disbursementSchedules[0].id} for student and ministry.`,
+          `Completed creating notifications for disbursement id: ${application.currentAssessment.disbursementSchedules[0].id} for student and ministry.`,
+        ]),
+      ).toBe(true);
+      const studentNotificationCount = await db.notification.count({
+        where: {
+          notificationMessage: {
+            id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+          },
+          metadata: {
+            disbursementId:
+              application.currentAssessment.disbursementSchedules[0].id,
+          },
+          user: { id: application.student.user.id },
+          creator: { id: systemUsersService.systemUser.id },
+        },
+      });
+      expect(studentNotificationCount).toBe(1);
+      const ministryNotificationCount = await db.notification.count({
+        where: {
+          notificationMessage: {
+            id: NotificationMessageType.MinistryNotificationDisbursementBlocked,
+          },
+          metadata: {
```

Comment:
Is it converting this to a valid postgres query with a jsonb condition ??

Reply:
Yes.

<img width="1451" alt="image" src="https://github.com/bcgov/SIMS/assets/7859295/a03eb93a-dd9d-438c-abc2-258f9c2aacfa">